modules/carto/src/layers/h3-tile-layer.ts

```diff
@@ -65,15 +65,35 @@ export default class H3TileLayer<DataT = any, ExtraPropsT = {}> extends Composit
 
   renderLayers(): Layer | null | LayersList {
     const {data, tileJSON} = this.state;
-    const maxZoom = parseInt(tileJSON.maxresolution);
+    let minresolution = parseInt(tileJSON.minresolution);
+    let maxresolution = parseInt(tileJSON.maxresolution);
+
+    // Convert Mercator zooms provided in props into H3 res levels
+    // and clip into valid range provided from the tilejson
+    if (this.props.minZoom) {
+      minresolution = Math.max(
+        minresolution,
+        getHexagonResolution({zoom: this.props.minZoom, latitude: 0})
+      );
+    }
+    if (this.props.maxZoom) {
+      maxresolution = Math.min(
+        maxresolution,
+        getHexagonResolution({zoom: this.props.maxZoom, latitude: 0})
+      );
+    }
+
+    // The naming is unfortunate, but minZoom & maxZoom in the context
+    // of a Tileset2D refer to the resolution levels, not the Mercator zooms
     return [
       new SpatialIndexTileLayer(this.props, {
         id: `h3-tile-layer-${this.props.id}`,
         data,
         // @ts-expect-error Tileset2D should be generic over TileIndex
         TilesetClass: H3Tileset2D,
         renderSubLayers,
-        maxZoom
+        minZoom: minresolution,
```

Comment:
```suggestion
        // minZoom and maxZoom are in h3 resolution, they will be converted to XYZ zoom levels by H3Tileset2D
        minZoom: minresolution,
```

Reply:
That's not quite right, H3Tileset2D will treat them as H3 resolutions, it is just the base Tileset2D class is hardcoded to treat zoom and resolution as equal

Reply:
ok, good to know :)